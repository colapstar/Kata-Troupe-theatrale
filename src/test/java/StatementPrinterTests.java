
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.List;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatementTXT() throws IOException, Exception {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet", new Tragedy("Hamlet"));
        plays.put("as-like", new Comedy("As You Like It"));
        plays.put("othello", new Tragedy("Othello"));

        Customer bigCo = new Customer("BigCo");

        Invoice invoice = new Invoice(bigCo, List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        statementPrinter.generateStatementData(invoice, plays);
        String result = statementPrinter.printTXT();

        verify(result);
    }

    @Test
    void exampleStatementHTML() throws IOException, Exception {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet", new Tragedy("Hamlet"));
        plays.put("as-like", new Comedy("As You Like It"));
        plays.put("othello", new Tragedy("Othello"));

        Customer bigCo = new Customer("BigCo");

        Invoice invoice = new Invoice(bigCo, List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        statementPrinter.generateStatementData(invoice, plays);
        statementPrinter.printHTML();

        String result = Files.readString(Paths.get("build/results/invoice.html"));

        verify(result);
    }
    @Test
    void promotionAppliedStatementTXT() throws IOException {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet", new Tragedy("Hamlet"));
        plays.put("as-like", new Comedy("As You Like It"));
        plays.put("othello", new Tragedy("Othello"));

        Customer bigCo = new Customer("BigCo");
        bigCo.fidelityPoints = 155; // Setting the fidelity points over 150 to apply the promotion

        Invoice invoice = new Invoice(bigCo, List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        statementPrinter.generateStatementData(invoice, plays);
        String result = statementPrinter.printTXT();

        verify(result);
    }

    @Test
    void promotionAppliedStatementHTML() throws IOException, Exception {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet", new Tragedy("Hamlet"));
        plays.put("as-like", new Comedy("As You Like It"));
        plays.put("othello", new Tragedy("Othello"));

        Customer bigCo = new Customer("BigCo");
        bigCo.fidelityPoints = 155; // Setting the fidelity points over 150 to apply the promotion

        Invoice invoice = new Invoice(bigCo, List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        statementPrinter.generateStatementData(invoice, plays);
        statementPrinter.printHTML();

        String result = Files.readString(Paths.get("build/results/invoice.html"));
        verify(result);
    }

}
