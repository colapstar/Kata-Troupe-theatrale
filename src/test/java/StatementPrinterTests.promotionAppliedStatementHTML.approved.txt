<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Invoice</title>
    <style>
        table, th, td {
            border: 1px solid black;
            border-collapse: collapse;
            padding: 10px;
            text-align: center;
        }
        table {
            width: 60%;
            margin: 20px 0;
        }
        th {
            background-color: #f2f2f2;
        }
    </style>
</head>
<body>
    <h2>Invoice</h2>
    <p><b>Client</b> : BigCo</p>

    <table>
<tr>
            <th>Piece</th>
            <th>Seats sold</th>
            <th>Price</th>
        </tr>
            <tr>
                <td>Hamlet</td>
                <td>55</td>
                <td>$650.00</td>
            </tr>
            <tr>
                <td>As You Like It</td>
                <td>35</td>
                <td>$580.00</td>
            </tr>
            <tr>
                <td>Othello</td>
                <td>40</td>
                <td>$500.00</td>
            </tr>
        <tr>
            <td colspan="2"><b>Total owed:</b></td>
            <td><b>$1,715.00</b></td>
        </tr>
        <tr>
            <td colspan="2"><b>Fidelity points earned:</b></td>
            <td><b>52</b></td>
        </tr>
    </table>
        <p>A promotion of $15 has been applied to your invoice due to your fidelity points.</p>
    <p style="color:red;">Payment is required under 30 days. <b>We can break your knees if you don't do so.</b></p>
</body>
</html>
